Index: UpgradeStartup.java
===================================================================
--- UpgradeStartup.java	(revision 56959)
+++ UpgradeStartup.java	(working copy)
@@ -314,7 +314,7 @@
             }
             logger.log(Level.INFO, "Repackaged application " + app.getName()
                 + " at " + repackagedFile.getPath()); 
-            deployParams.path = repackagedFile;
+            deployParams.setPath(repackagedFile);
         }
 
         deployParams.properties = app.getDeployProperties();
