Index: ArchiveFactory.java
===================================================================
--- ArchiveFactory.java	(revision 56959)
+++ ArchiveFactory.java	(working copy)
@@ -218,7 +218,20 @@
             throw new MalformedURLException("Protocol not supported : " + provider);
         } 
     }
-    
+
+    public ReadableArchive openArchive(URI uri, DeployCommandParameters properties) throws IOException {
+        for (ReadableArchiveFactory fac : habitat.<ReadableArchiveFactory>getAllServices(ReadableArchiveFactory.class)) {
+            //get the first ReadableArchive and move
+            try{
+                ReadableArchive archive = fac.open(uri, properties);
+                if(archive != null) return archive;
+            }catch(Exception e){
+                //ignore?
+            }
+        }
+        return openArchive(uri);
+    }
+
     /**
      *Create a URI for the jar specified by the path string.
      *<p>
