Index: EJBContainerImpl.java
===================================================================
--- EJBContainerImpl.java	(revision 56959)
+++ EJBContainerImpl.java	(working copy)
@@ -57,6 +57,7 @@
 import com.sun.appserv.connectors.internal.api.ConnectorRuntime;
 import com.sun.enterprise.util.io.FileUtils;
 
+import org.glassfish.api.deployment.DeployCommandParameters;
 import org.glassfish.embeddable.Deployer;
 import org.glassfish.embeddable.GlassFish;
 import org.glassfish.embeddable.GlassFishException;
@@ -124,7 +125,7 @@
 
             // Check if appName was set by application creation code
             appName = res_app.getAppName();
-            
+            DeployCommandParameters dp = new DeployCommandParameters();
             String[] params;
             if (appName != null) {
                 params = new String[] {"--name", appName};
@@ -137,6 +138,7 @@
                 _logger.info("[EJBContainerImpl] Deploying as a ScatteredArchive");
                 deployedAppName = deployer.deploy(((ScatteredArchive)app).toURI(), params);
             } else {
+                dp.setPath((File)app);
                 _logger.info("[EJBContainerImpl] Deploying as a File");
                 deployedAppName = deployer.deploy((File)app, params);
             }
