Index: ValidateRemoteDirDeploymentCommand.java
===================================================================
--- ValidateRemoteDirDeploymentCommand.java	(revision 57619)
+++ ValidateRemoteDirDeploymentCommand.java	(working copy)
@@ -168,7 +168,7 @@
          */
         final ParameterMap paramMap = new ParameterMap();
         paramMap.add("checksum", Long.toString(checksum));
-        paramMap.add("DEFAULT", path.toURI().getSchemeSpecificPart());
+        paramMap.add("DEFAULT", getPath().getSchemeSpecificPart());
 
         ActionReport.ExitCode replicateResult = ClusterOperationUtil.replicateCommand(
                 "_instanceValidateRemoteDirDeployment",
@@ -193,14 +193,14 @@
      */
     private ReadableArchive archive(final Logger logger, final ActionReport report) {
         try {
-            return archiveFactory.openArchive(path, this);
+            return archiveFactory.openArchive(getPath(), this);
         } catch (IOException e) {
             final String msg = localStrings.getLocalString("deploy.errOpeningArtifact",
-                    "deploy.errOpeningArtifact", path.getAbsolutePath());
+                    "deploy.errOpeningArtifact", getPathAsFile().getAbsolutePath());
             if (logReportedErrors) {
                 report.failure(logger, msg, e);
             } else {
-                report.setMessage(msg + path.getAbsolutePath() + e.toString());
+                report.setMessage(msg + getPathAsFile().getAbsolutePath() + e.toString());
                 report.setActionExitCode(ActionReport.ExitCode.FAILURE);
             }
             return null;
