Index: MapInjectionResolver.java
===================================================================
--- MapInjectionResolver.java	(revision 57265)
+++ MapInjectionResolver.java	(working copy)
@@ -45,6 +45,7 @@
 import java.lang.reflect.AnnotatedElement;
 import java.lang.reflect.Field;
 import java.lang.reflect.Type;
+import java.net.URI;
 import java.security.AccessController;
 import java.security.PrivilegedAction;
 import java.util.ArrayList;
@@ -388,10 +389,29 @@
                 convertStringToStringArray(paramValStr, param.separator());
         } else if (type.isAssignableFrom(File.class)) {
             return new File(paramValStr);
+        } else if (type.isAssignableFrom(URI.class)) {
+            paramValStr = ConvertStringToURIMode(paramValStr);
+            return URI.create(paramValStr);
         }
         return paramValue;
     }
 
+    private static String ConvertStringToURIMode(String paramValStr){
+        int index;
+        String newparamValStr;
+        if (paramValStr.contains("file:\\")){
+            index = paramValStr.indexOf("file:\\");
+            newparamValStr = paramValStr.substring(index);
+            newparamValStr = newparamValStr.replace("\\", "/");
+        } else if (paramValStr.contains("file:/")){
+            return paramValStr;
+        } else {
+            File file = new File(paramValStr);
+            newparamValStr = file.toURI().toString();
+        }
+        return newparamValStr;
+    }
+
     /**
      * Convert the List<String> parameter to the specified type.
      *
